### PR TITLE
Adds arrays without size section to the arrays doc page

### PR DESCRIPTION
**Pull request by damirka**
Closes https://github.com/AleoHQ/leo/issues/1320

### CODE REVIEW

File: documentation/developer/language/04_arrays_and_tuples.md [hunk 1]
```diff
@@ -72,6 +72,30 @@ Leo supports static arrays with fixed size.
 **Dynamic arrays do not exist in Leo**. 
 You cannot change the length of an array once it is initialized.
 
+### Arrays without size
```
Comment (acoglio) [inline documentation/developer/language/04_arrays_and_tuples.md:75]: Arrays without size => Arrays with Unspecified Size

(Reason: arrays always have a size, we just allow it to be left unspecified.)

File: documentation/developer/language/04_arrays_and_tuples.md [hunk 2]
```diff
@@ -72,6 +72,30 @@ Leo supports static arrays with fixed size.
 **Dynamic arrays do not exist in Leo**. 
 You cannot change the length of an array once it is initialized.
 
+### Arrays without size 
+Leo allows developers to define arrays, sizes of which will be defined at compile time. Arrays with unknown
```
Comment (acoglio) [inline documentation/developer/language/04_arrays_and_tuples.md:76]: define arrays, sizes of which will be defined at compile time
=>
define arrays without specifying the sizes, which are determined by the compiler

File: documentation/developer/language/04_arrays_and_tuples.md [hunk 3]
```diff
@@ -72,6 +72,30 @@ Leo supports static arrays with fixed size.
...
+size are marked with `_` (underscore):
```
Comment (acoglio) [inline documentation/developer/language/04_arrays_and_tuples.md:77]: with unknown size are marked
=>
with unspecified size are marked

**Review by acoglio — APPROVED** [review]
Looks good, I've added some comments for consideration.

**Review by gluax — APPROVED** [review]
LGTM I agree with @acoglio's suggestions.

**Review by collinc97 — APPROVED** [review]

**Pull request merged**